crates/polars-time/src/chunkedarray/rolling_window/rolling_kernels/no_nulls.rs

```diff
@@ -99,25 +99,29 @@ where
                 // `by`, which has already been checked to be the same length as the values.
                 unsafe { *out.get_unchecked_mut(*out_idx as usize) = res };
             } else {
-                null_positions.push(*out_idx as usize)
+                if validity.is_none() {
```

Comment:
Maybe put this in a closure as that ensure we have a single code block and call that closure twice.

I think the overhead of the function call is not too much here as you said, you don't expect many nulls.